BacktrackerBenchmarks/BacktrackerCThree.cs

```diff
@@ -0,0 +1,179 @@
+using System.Diagnostics.CodeAnalysis;
+using Sudoku;
+using Puzzle = PuzzleQuick.Puzzle;
+
+namespace BacktrackerThree;
+
+/*
+    Backtracker, based on array, collection, span, and integer data types.
+    Adds use of pre-computed data, spans, bit twiddling, and ref ints relative to baseline.
+    Relies on a helper class for some of the more complicated Sudoku logic.
+
+    This implementation is built on the following premises:
+
+    - We can represent the units (rows, columns, boxes) as a set of 9 cell lists, with legal values 0-9.
```

Comment:
Transforming the data to a representation that allows more efficient execution is a common trick used by more complicated algorithms.

I think there are even better data representations. One possible improvement is to use stride of 4 instead of 3 to avoid multiple and divide by 3 operations, at the cost of some cells being unused. Multiply and divide by 4 is much faster compared to multiply and divide by 3.